Added new kb article grid-handle-empty-popup-footer (#2414)
telerik/blazor-docs: knowledge-base/grid-handle-empty-popup-footer.md

---

> title:  How to handle the empty popup footer when using the Popup Form Template

Reviewer: * "Handling" = "How to handle" but shorter
* Use title case
```suggestion
title:  Handling the Empty Popup Footer when Using the Popup Form Template
```

Reviewer 2: In addition, "handle" seems too generic. Why not use a keyword that a customer would search with, e.g. "hide" or "remove"

Reviewer 3: Updated in https://github.com/telerik/blazor-docs/pull/2414/commits/783d5e16f8d98b77f5d5d01c81fe7708f1fc69ba

> title:  How to handle the empty popup footer when using the Popup Form Template

Reviewer 3: ```suggestion
title:  Hiding or Removing the Empty Popup Footer when Using the Popup Form Template
```

The requirement is to either hide this empty footer or utilize it by placing the custom form buttons within. 

This KB article also answers the following questions:

- How to hide the empty footer in the edit popup of the Grid for Blazor?
- How to display custom buttons in the footer of the edit popup?
- How to manage form submission with custom buttons in the Grid for Blazor?

## Solution

There are two approaches to address this requirement:

* [Option 1: Display Custom Buttons in the Footer](#option-1-display-custom-buttons-in-the-footer)
* [Option 2: Remove the Footer and Keep the Buttons in the FormTemplate](#option-2-remove-the-footer-and-keep-the-buttons-in-the-formtemplate)

### Option 1: Display Custom Buttons in the Footer

To display custom buttons in the footer and handle form submission, follow these steps:

1. Declare a [`<FormButtons>`]({%slug form-formitems-buttons%}) tag inside the custom Form and leave it empty, so the Form does not render its default buttons.
2. Declare custom buttons in the [`<ButtonsTemplate>`]({%slug grid-templates-popup-buttons%}) and handle their `OnClick` events to manage the Form submission.

> >caption Here is an example demonstrating how to display custom buttons in the popup footer

Reviewer: Captions should resemble titles instead of introductory sentences.
```suggestion
>caption Displaying custom buttons in the popup footer
```

    {
        var index = People.FindIndex(i => i.EmployeeId == person.EmployeeId);
        if (index != -1)
        {
            People[index] = person;
        }
    }

    private void CreatePerson(Person person)
    {
        person.EmployeeId = People.Max(x => x.EmployeeId) + 1;

        People.Insert(0, person);
    }

    private List<Person> GeneratePeople(int count, int startIndex = 0)
    {
        List<Person> result = new List<Person>();

        for (int i = startIndex; i < startIndex + count; i++)
        {
            result.Add(new Person()
                {
                    EmployeeId = i,
                    Name = "Employee " + i.ToString(),
                    HireDate = new DateTime(2020, 6, 1).Date.AddDays(count - (i % 7)),
                    Position = i % 3 <= 2 ? PositionsData[i % 3] : PositionsData.FirstOrDefault()

                });
        }

        return result;
    }
    #endregion
}
````

### Option 2: Remove the Footer and Keep the Buttons in the FormTemplate

This approach relies on using CSS to hide the empty footer. Add your custom class to the edit popup of the Grid to override its default styling.

> >caption Here is an example demonstrating how to hide the empty popup footer

Reviewer: ```suggestion
>caption Hiding the empty popup footer
```
